lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_integration.rb

```diff
@@ -30,6 +35,19 @@ class GeoEngineer::Resources::AwsApiGatewayIntegration < GeoEngineer::Resource
 
   after :initialize, -> { _terraform_id -> { NullObject.maybe(remote_resource)._terraform_id } }
 
+  # Override this method so we can ensure the "type" field does get set
+  # correctly before data is sent to geoengineer.
+  def terraform_attributes
+    result = super
+
+    # Add the expected "type" attribute
+    result['type'] = result['integration_type']
```

Comment:
So this is sure to not override the `type` field from Geoengineer::Resource? If so, it LGTM, might want a test to verify that though.